### PR TITLE
accesscontrol service read replica

### PR DESCRIPTION
**Pull request by mildwonkey**
This PR has a corresponding enterprise PR: https://github.com/grafana/grafana-enterprise/pull/6832

This PR refactors the accesscontrol service to take a `db.ReplDB` in place of the `db.DB` interface and modifies the access control store to use the `db.DB()` or `db.ReadReplica()` depending on the sql. This will have no effect in Grafana instances that don't use the readReplica feature toggle. If the toggle is on, but the replica isn't configured, Grafana will return the primary `DB()` in place of the `ReadReplica()`.

### CODE REVIEW

**Comment by mildwonkey** [comment]
/deploy-to-hg --enterprise-ref mildwonkey/accesscontrol-read-repl

File: pkg/services/accesscontrol/database/database.go [hunk 1]
```diff
@@ -36,17 +36,17 @@ const (
...
 
 type AccessControlStore struct {
-	sql db.DB
+	sql db.ReplDB
 }
 
 func (s *AccessControlStore) GetUserPermissions(ctx context.Context, query accesscontrol.GetUserPermissionsQuery) ([]accesscontrol.Permission, error) {
 	result := make([]accesscontrol.Permission, 0)
-	err := s.sql.WithDbSession(ctx, func(sess *db.Session) error {
+	err := s.sql.DB().WithDbSession(ctx, func(sess *db.Session) error {
```
Comment (kalleep) [inline pkg/services/accesscontrol/database/database.go:49]:
```suggestion
	err := s.sql.ReadReplica().WithDbSession(ctx, func(sess *db.Session) error {
```

Comment (kalleep) [reply pkg/services/accesscontrol/database/database.go:49]: If the idea here is to always prefer read replica for read this should probably change right?

**Review by kalleep — COMMENTED** [review]
Looks good to me!

>let's leave zanzana out of this, it didn't do anything to me

:rofl: that is mostly throwaway code that we will replace anyway, we just have it there for now so we can start testing stuff out

**Review by kalleep — APPROVED** [review]

File: pkg/services/accesscontrol/database/database.go (hunk 1 above)
Comment (mildwonkey) [reply pkg/services/accesscontrol/database/database.go]: Yes indeed - great catch, thank you!

**Review by jalevin — APPROVED** [review]

**Pull request merged**